File: ios/ReactNativeDemo/PingPongModule.m
#import "PingPongModule.h"
#import <React/RCTLog.h>

@implementation PingPongModule

RCT_EXPORT_MODULE();

RCT_EXPORT_METHOD(send:(NSString *)input
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
{
    RCTLogInfo(@"Received input: %@", input);
    resolve(@"bye from ios");
}

@end

File: android/app/src/main/java/com/reactnativedemo/PingPongModule.kt
package com.reactnativedemo

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.Promise
import android.util.Log

class PingPongModule(reactContext: ReactApplicationContext) :
    ReactContextBaseJavaModule(reactContext) {
    override fun getName() = "PingPongModule"

    @ReactMethod
    fun send(input: String, promise: Promise) {
        Log.d("PingPongModule", "Received input: $input")
        try {
            promise.resolve("bye from android")
        } catch (e: Exception) {
            promise.reject("ERROR", e.message, e)
        }
    }
}
